src/js/tech/middleware.js

```diff
@@ -82,6 +82,20 @@ export function setSource(player, src, next) {
   player.setTimeout(() => setSourceHelper(src, middlewares[src.type], next, player), 1);
 }
 
+/**
+ * Allows user to override the returned value of player.currentSource() using middleware.
+ *
+ * @param {Tech~SourceObject} src
+ *         A source object.
+ * @param {Object[]} middleware
+ *        An array of middleware instances.
+ * @return {Tech~SourceObject}
+ *         A source object.
+ */
+export function currentSource(src, middleware) {
```

Comment:
thought: I was thinking about whether this should be a separate method or if `middleware.get` should be able to handle this. Though, I guess the difference is here we don't want it to actually get to the tech? Though, maybe that's that it gets it from the tech, but potentially we'd want a more generic `get` that can also accept a default value?